pkg/assembler/backends/testing/backend.go

```diff
@@ -157,6 +159,44 @@ func (c *demoClient) Builders(ctx context.Context, builderSpec *model.BuilderSpe
 	return builders, nil
 }
 
+func (c *demoClient) HashEquals(ctx context.Context, hashEqualSpec *model.HashEqualSpec) ([]*model.HashEqual, error) {
+	var hashEquals []*model.HashEqual
+
+	for _, h := range c.hashEquals {
+		if hashEqualSpec.Justification == nil || h.Justification == *hashEqualSpec.Justification {
```

Comment:
can you turn these into early returns or set temp flags instead of nested ifs?

e.g.

```
if hashEqualSpec.Justification != nil || h.Justification != *hashEqualSpec.Justification {
  return hashEquals, nil
}
if ... {
  return hashEqual, nil
```

or 

```
if hashEqualSpec.Justification == nil || h.Justification == *hashEqualSpec.Justification {
   justification = true
}
if ... {
   blah = true
...

if justification && blah && ... { ...}
```